File: admin_app/admin_flutter/lib/models/website_contact_request.dart
class WebsiteContactRequest {
  const WebsiteContactRequest({
    required this.id,
    required this.name,
    required this.phone,
    required this.status,
    required this.source,
    required this.createdAt,
    required this.updatedAt,
    this.email,
    this.companyName,
    this.message,
  });

  final int id;
  final String name;
  final String phone;
  final String? email;
  final String? companyName;
  final String? message;
  final String status;
  final String source;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory WebsiteContactRequest.fromJson(Map<String, dynamic> json) {
    return WebsiteContactRequest(
      id: json['id'] as int,
      name: json['name'] as String,
      phone: json['phone'] as String,
      email: json['email'] as String?,
      companyName: json['company_name'] as String?,
      message: json['message'] as String?,
      status: json['status'] as String,
      source: json['source'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }
}

File: backend/app/models/website_contact_request.py
from datetime import datetime

from sqlalchemy import DateTime, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.master_database import MasterBase


class WebsiteContactRequest(MasterBase):
    __tablename__ = "website_contact_requests"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(150), nullable=False)
    phone: Mapped[str] = mapped_column(String(40), nullable=False, index=True)
    email: Mapped[str | None] = mapped_column(String(180))
    company_name: Mapped[str | None] = mapped_column(String(180))
    message: Mapped[str | None] = mapped_column(Text)
    status: Mapped[str] = mapped_column(
        String(30),
        default="new",
        server_default="new",
        index=True,
    )
    source: Mapped[str] = mapped_column(
        String(40),
        default="institutional_site",
        server_default="institutional_site",
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
    )
